src/vm/threadsuspend.cpp

```diff
@@ -5322,6 +5328,30 @@ HRESULT ThreadSuspend::SuspendRuntime(ThreadSuspend::SUSPEND_REASON reason)
             }
 #endif
 
+#if defined(FEATURE_HIJACK) && defined(PLATFORM_UNIX)
+            _ASSERTE (thread == NULL);
+            while ((thread = ThreadStore::GetThreadList(thread)) != NULL)
+            {
+                if (thread == pCurThread)
+                    continue;
+
+                if ((thread->m_State & Thread::TS_GCSuspendPending) == 0)
+                    continue;
+
+                if (!thread->m_fPreemptiveGCDisabled)
+                    continue;
+
+                // When we tried to inject the suspension before, we may have been in a place
+                // where it wasn't possible. Try one more time.
+                bool gcSuspensionSignalSuccess = thread->InjectGcSuspension();
+                if (!gcSuspensionSignalSuccess)
+                {
+                    // If we failed to raise the signal for some reason, just log it and move on.
+                    STRESS_LOG1(LF_SYNC, LL_INFO1000, "Thread::SuspendRuntime() -   Failed to raise GC suspension signal for thread %p.\n", thread);
+                }
+            }
+#endif
+
 #ifndef DISABLE_THREADSUSPEND
```

Comment:
It seems it would be nice to merge the loop above and below and just have the actual suspension mechanism ifdefed. And maybe even better, refactor the code to extract the suspension mechanism into a separate function(s).